### ISSUE TITLE
Posting code is bugged

### ISSUE THREAD

**Issue by cyberalien** [elk-zone/elk#917]
Example of post text (link instead of code because GitHub converts Markdown): https://gist.github.com/cyberalien/3bf1871bc10236acb5c09c394d8d5422

When posting that message (copy/paste from Gist), editor bugs out, treating whole message as code block:

![posting-bug](https://user-images.githubusercontent.com/822287/211390191-46e9b4fb-7ff5-497c-9a77-cb1a3509975c.png)

This is result of that post in Mastodon and Elk (I'll open a different bug report for incorrectly rendering it):

![posting-result-mastodon](https://user-images.githubusercontent.com/822287/211391046-56fd3b7b-1235-495a-bbf0-5a1b2a2f443c.png)

![post-bug-result](https://user-images.githubusercontent.com/822287/211391073-2a60f3e7-613b-4679-af67-4c3189a8050b.png)

**Comment by cyberalien** [comment]
> This should now be resolved. Let me know if not and we can reopen.

Still bugged. I think you closed wrong issue. Other issues were about rendering that code, which was also bugged. This issue is about editor.

**Comment by danielroe** [comment]
I was able to successfully post that code 👀 Are you still not able to do so?

**Comment by cyberalien** [comment]
No. Tried it on local from latest `main` branch, tried it on `elk.zone`. In both cases editor shows exactly the same bugged sample as in first screenshot.

Result is a bit different, now rendered correctly for what editor generated, but still bugged because editor generates wrong code by adding ``` before and after pasted message.

**Comment by cyberalien** [comment]
Result demo: https://elk.zone/mas.to/@cyberalien/109675703583251205

**Comment by cyberalien** [comment]
Did that, still bugged.

But something weird just happened. I tried recording demo with a clean browser (purged stuff, logged in, started demo) and this happened:
https://user-images.githubusercontent.com/822287/212043051-30059d35-9294-4a65-a057-dfec7ed232fb.mov

I'm not sure what's going on here. After posting message, Elk went in infinite loop. Restarted vite, reloaded page, purged all site data and cache, as soon as I log in to that account, it freezes again, just like at the end of this video.

However, it was enough to show bug. In your demo when you pasted code, editor rendered it correctly. In my demo, as you can see from 0:05, it pasted text incorrectly.

**Comment by cyberalien** [comment]
Figured it out. It wasn't infinite loop, it was an uncaught exception that prevented rendering.

In `htmlToText()` in `content-parse.ts`, function `parse()` threw exception when trying to parse that post.

Dumped `html` variable = `"</p><p>"`.

Not related to this bug, but caused by this bug. Post content passed to `parseMastodonHTML()` is this:

```
<p>```<br />Inline code with link: `<a href=\"https://api.iconify.design/noto.css?icons=1st-place-medal,2nd-place-medal\" target=\"_blank\" rel=\"nofollow noopener noreferrer\"><span class=\"invisible\">https://</span><span class=\"ellipsis\">api.iconify.design/noto.css?ic</span><span class=\"invisible\">ons=1st-place-medal,2nd-place-medal</span></a>`</p><p>HTML block code:<br />```html<br />&lt;span class=&quot;icon--noto icon--noto--1st-place-medal&quot;&gt;&lt;/span&gt;<br />&lt;span class=&quot;icon--noto icon--noto--2nd-place-medal-medal&quot;&gt;&lt;/span&gt;<br />```</p><p>```</p>
```

**Comment by cyberalien** [comment]
Did a bit more debugging. It looks like source of error is TipTap. 

Added debug to onUpdate() event
```js
onUpdate({ editor }) {
      const value = editor.getHTML()
      content.value = value
      console.log('onUpdate:', JSON.stringify(value))
    }
```

When posting simple text, it logs this:
```
onUpdate: "<p>Test</p>"
```
When pasting code sample linked in first post, it logs this:
```
onUpdate: "<pre><code>Inline code with link: `https://api.iconify.design/noto.css?icons=1st-place-medal,2nd-place-medal`\n\nHTML block code:\n```html\n&lt;span class=\"icon--noto icon--noto--1st-place-medal\"&gt;&lt;/span&gt;\n&lt;span class=\"icon--noto icon--noto--2nd-place-medal-medal\"&gt;&lt;/span&gt;\n```</code></pre>"
```
So it wrapped whole post in `<code></code>` instead of `<p></p>`

**Comment by cyberalien** [comment]
Opened issue at Tiptap: https://github.com/ueberdosis/tiptap/issues/3627

**Comment by danielroe** [comment]
Or maybe the issue is that what is being copied into your clipboard includes that `<code>`? Perhaps a feature of GitHub?

**Comment by cyberalien** [comment]
When pasting it to any editor, it doesn't include anything else. I'm copying from raw file: https://gist.githubusercontent.com/cyberalien/3bf1871bc10236acb5c09c394d8d5422/raw/86e228c5ab568161378ead7ffed3e6aa822906d1/test.txt

Actually, forgot this very important step in reproduction instructions: click "Raw" button before copying code to get raw file.

**Comment by zaidhaan** [comment]
Just to add on a bit with a few things that I know myself, I believe that danielroe is correct about the clipboard adding pre/code tags as a feature of GH/browser (https://github.com/ueberdosis/tiptap/issues/3627#issuecomment-1452267977). If you copy the raw string contents that you mentioned, you should get the following pasted into elk:
![image](https://user-images.githubusercontent.com/5954994/222512259-e94d8845-5b7e-425a-9569-c80a54c24c68.png)

Post that and all is good:
![image](https://user-images.githubusercontent.com/5954994/222512454-b1178459-93a1-42d3-9e5c-08a683fbe141.png)

Only issue with this maybe might be that the HTML code formatting isn't automatically applied on paste in the preview.

If you refer to my linked comment, at the end I noted how a few platforms seem to handle the whole added formatting situation properly anyway, so hopefully tiptap does so too in the near future, which would cause text pasted to be interpreted correctly on the tiptap front, which would result in proper formatting in elk (as shown in my first screenshot).
